partiql-tests-data/eval/primitives/operators/like.ion

```diff
@@ -0,0 +1,1624 @@
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+      ]
+    }
+  },
+  {
+    name:"like-expression with ESCAPE specified that DO NOT match{text:\" '%'    \",pattern:\" '[%[%'   \"}",
+    statement:"SELECT * FROM `[true]` WHERE '%' LIKE '[%[%' ESCAPE '['",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+      ]
+    }
+  },
+  {
+    name:"like-expression with ESCAPE specified that DO NOT match{text:\" '100%' \",pattern:\" '1000[%' \"}",
+    statement:"SELECT * FROM `[true]` WHERE '100%' LIKE '1000[%' ESCAPE '['",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+      ]
+    }
+  },
+  {
+    name:"like-expression with ESCAPE specified that DO NOT match{text:\" '100%' \",pattern:\" '[%100'  \"}",
+    statement:"SELECT * FROM `[true]` WHERE '100%' LIKE '[%100' ESCAPE '['",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+      ]
+    }
+  },
+  {
+    name:"like-expression with ESCAPE specified that DO NOT match{text:\" '100%' \",pattern:\" '1_[%_'  \"}",
+    statement:"SELECT * FROM `[true]` WHERE '100%' LIKE '1_[%_' ESCAPE '['",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+      ]
+    }
+  },
+  {
+    name:"like-expression with ESCAPE specified that DO NOT match{text:\" '100%' \",pattern:\" '1%_[%1' \"}",
+    statement:"SELECT * FROM `[true]` WHERE '100%' LIKE '1%_[%1' ESCAPE '['",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+      ]
+    }
+  },
+]
+
+'empty-string-for-like-pattern'::[
+  envs::{
+    people:[
+      {
+        first_name:"John",
+        last_name:"Doe"
+      }
+    ]
+  },
+  {
+    name:"Empty String for LIKE pattern evaluates to false",
+    statement:"SELECT * FROM people AS p WHERE p.first_name LIKE ''",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+      ]
+    }
+  },
+  {
+    name:"Empty String for LIKE pattern and text evaluates to true",
+    statement:"SELECT * FROM people AS p WHERE '' LIKE ''",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+        {
+          first_name:"John",
+          last_name:"Doe"
+        }
+      ]
+    }
+  },
+  {
+      name:"More than one character given for ESCAPE",
+      statement:"SELECT * FROM `[true]` WHERE 'a' LIKE 'a' ESCAPE 'aa'",
```

Comment:
This was the somewhat ambiguous case for an invalid function argument that's not due to typing error -- hence why the behavior is the same for both eval modes. PostgreSQL don't do a "smarter" static check for the escape pattern of a `LIKE` operator but gives an evaluator error.